doc/MATH.rst

```diff
@@ -0,0 +1,162 @@
+
+   cmplx aim = cnormalize(global.plr.pos - enemy->pos);
+
+This ``aim`` variable could be passed to a ``move_towards`` function attached
+to a ``PROJECTILE`` object. The effect here would be the bullets moving
+directly towards the player in a straight line, even if the player moves around
+the screen, following the player as they move around in real-time (assuming)
+more than one bullet is fired.
+
+Let's look at the argument inside ``cnormalize`` first, ``global.plr.pos -
+e->pos``.  Both ``global.plr.pos`` and ``e->pos`` are *complex numbers*, in
+that they have both *real* and *imaginary* parts. Much like the example in
+``Simple Movement``, they represent a place on the X/Y grid.
+
+In the format of ``[X,Y]``, let's say that ``global.plr.pos`` is ``[-1, 6]``,
+and that ``enemy->pos`` is ``[6, 3]``.
+
+.. image:: images/math-01.png
+   :width: 300pt
+
+When you subtract ``[6, 3]`` (enemy position) from ``[-1, 6]`` (player
+position), you end up with ``[-7, 3]``, as seen here with ``plr->pos``.
+
+.. image:: images/math-02.png
+   :width: 300pt
+
+This also conveniently lets the enemy position ``enemy->pos`` become the new
+"origin," or ``[0, 0]``. This is useful because it means that we can more
+easily determine what angle the danmaku need to travel in to travel towards the
+player.
+
+Since we don't really care about the distance, as we're looking for an angle
+or direction instead, ``cnormalize()`` in the original code mostly ignores
+that so that we get the complex number between ``[0, 0]`` and ``[-7, 3]``.
+You can see the formula used for converting a *cartesean coordinate* to a
+*polar coordinate* `here
+<https://www.engineeringtoolbox.com/converting-cartesian-polar-coordinates-d_1347.html>`__,
+but for the sake of brevity, the angle between the two, assuming **0°** is
+"positive" on the X-axis, is **157°**, which you can see from the orange line
+rotating counter-clockwise to about **157°** on the second image.
+
+This could still be done, technically, using traditional vectors. However,
+there are still significant advantages to doing it this way. Let's consider how
+we might use this new ``aim`` variable later on, say in a ``PROJECTILE`` block
+for a danmaku bullet:
+
+.. code-block:: c
+
+   // aim directly at the player
+   cmplx aim = cnormalize(global.plr.pos - enemy->pos);
+
+   // a bit of randomization
+   cmplx offset = cdir(M_PI/180 * rng_sreal());
+
+   // later, inside a PROJECTILE() block...
+   .move = move_asymptotic_simple(aim * offset, 5),
+
+The important piece here is the ``aim * offset`` inside the ``move()`` block.
+Being able to multiply complex numbers by each other means "procedurally"
+generating danmaku patterns becomes much easier. Multiplying two complex
+numbers together like this means their **polar coordinates** are multiplied,
+and in the case of something like ``cdir(M_PI/180 * rng_sreal())``, you can add
+"predictable randomization" to your patterns without having to wrangle with
```

Comment:
Why is this randomization predictable?

vector matricies -> matrices

-> You can quickly do rotations in your patterns without handling cumbersome matrices. In this case, we add some random scattering to the original direction of […]